lib/plugins.py

```diff
@@ -514,12 +516,13 @@ def __init__(self, parent, config, name):
         self.name = name
         self.config = config
         self.wallet = None
+        self._hooks_i_registered = []
         # add self to hooks
-        for k in dir(self):
-            if k in hook_names:
-                l = hooks.get(k, [])
-                l.append((self, getattr(self, k)))
-                hooks[k] = l
+        for aname in dir(self):
+            func = getattr(self, aname, None)
```

Comment:
As I mentioned, my worry is that someone might do:
```
class Plugin(BasePlugin):
    ...
    @property
    def my_cached_attr(self,):
        <do thing that only is valid after init_qt has been invoked>
```
This getattr will access *all* attributes, which includes calling this `my_cached_attr`.

Reply:
Ah. True. Is there a way to guard against that? Like find out if a thing is a property before calling `getattr`?  Maybe clever use of either `__getattr__` or `__getattribute__`? (I know they have a distinction which escapes me now)...?

Reply:
Oh. I know how. Just query the class. Will fix now.

Reply:
Ah! Durrr.... I should have thought of that lol. No need for metaclasses if we do that.

Reply:
The metaclass solution is only really better if we are making multiple instances of the Plugin object and we don't do much inheritance. The scanning of class during init is better if there are fewer instances than inheritances (which is more realistic for plugins).

Reply:
Ok.. added a commit to deal with the @property situation.

Note if you have ideas about doing it during plugin init itself that's cool too. This PR was just a quick fix to the existing mechanism.

Note that before we merge any fix we really should just make sure existing plugins won't break (just have to manually run through the code for all extant plugins against a master hook name list.. :)  )

And in the case where some odd plugin **DIDN'T** use `@hook`, but expected "hook-like-behavior", just add the missing `@hook`...